src/tools/rbd_mirror/image_replayer/BootstrapRequest.cc

```diff
@@ -607,16 +363,11 @@ void BootstrapRequest<I>::handle_image_sync(int r) {
     std::lock_guard locker{m_lock};
     m_image_sync->put();
     m_image_sync = nullptr;
+  }
 
-    if (m_canceled) {
-      dout(10) << "request canceled" << dendl;
-      m_ret_val = -ECANCELED;
-    }
-
-    if (r < 0) {
-      derr << "failed to sync remote image: " << cpp_strerror(r) << dendl;
-      m_ret_val = r;
-    }
+  if (r < 0) {
```

Comment:
nit: I think the check for `m_canceled` protected from a potential race when `m_image_sync->cancel()` was called too late to make the image sync request return the "canceled" error code. It looks like it does not hurt though and your change is ok.

But it seems like we unintentionally (in the original version and in yours) output the error message for the case when the image sync request is cancelled. May be fix this while here?

Reply:
No longer log it via `derr`